app/code/Magento/InventorySales/Plugin/Store/WebsiteResourcePlugin.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Plugin\Store;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventorySales\Model\AssignWebsiteToDefaultStock;
+use Magento\InventorySales\Model\ResourceModel\UpdateSalesChannelWebsiteCode;
+
+class WebsiteResourcePlugin
+{
+    /**
+     * @var UpdateSalesChannelWebsiteCode
+     */
+    private $updateSalesChannelWebsiteCode;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var AssignWebsiteToDefaultStock
+     */
+    private $assignWebsiteToDefaultStock;
+
+    /**
+     * WebsiteResourcePlugin constructor.
+     * @param UpdateSalesChannelWebsiteCode $updateSalesChannelWebsiteCode
+     * @param AssignWebsiteToDefaultStock $assignWebsiteToDefaultStock
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        UpdateSalesChannelWebsiteCode $updateSalesChannelWebsiteCode,
+        AssignWebsiteToDefaultStock $assignWebsiteToDefaultStock,
+        ResourceConnection $resourceConnection
+    ) {
+        $this->updateSalesChannelWebsiteCode = $updateSalesChannelWebsiteCode;
+        $this->resourceConnection = $resourceConnection;
+        $this->assignWebsiteToDefaultStock = $assignWebsiteToDefaultStock;
+    }
+
+    /**
+     * Get code from database
+     * @param int $websiteId
+     * @return string
+     */
+    private function getCodeFromDatabase(int $websiteId): string
+    {
+        $connection = $this->resourceConnection->getConnection();
+        $tableName = $connection->getTableName('store_website');
+        $selectQry = $connection->select()->from($tableName, 'code')->where('website_id = ?', $websiteId);
+        return (string) $connection->fetchOne($selectQry);
```

Comment:
I'm not sure that we need to return `''`(empty string) if the record was not found (website id is not existing)
More clear is return null in this case

Reply:
That method can't return an empty string. If you call it you call from `aroundSave` and only if the website exists.

Maybe I should change the initialization of `$oldCode = '';` to `$oldCode = null;`.